src/routes/auth.js

```diff
@@ -468,7 +467,48 @@ router.get('/verifyRToken', async (req, res) => {
             msg: 'User authenticated successfully',
         });
     } catch (error) {
-        console.log(error);
+        return res.status(401).json({
+            err: true,
+            msg: 'Unauthorized Client',
+        });
+    }
+});
+
+router.post('/getToken', async (req, res) => {
+    try {
+        const { token } = req.body;
+        const { requestToken } = decode(token);
+        rtoken.exists = util.promisify(rtoken.exists);
+        rtoken.hget = util.promisify(rtoken.hget);
+        const exists = await rtoken.exists(requestToken.toString());
+        if (!exists) {
+            return res.status(401).json({
+                err: true,
+                msg: 'Session Expired',
+            });
+        }
+
+        const clientId = await rtoken.hget(requestToken.toString(), 'cId');
+        const userId = await rtoken.hget(requestToken.toString(), 'uId');
+        const client = await Client.findById(clientId);
+
+        if (!client) {
+            return res.status(400).json({
+                err: true,
+                msg: 'No client found',
+            });
+        }
+
+        verify(requestToken, keys.privateKey, {
+            algorithms: ['HS256'],
+        });
+
```

Comment:
The verification is to be done here, after `requestToken` and `client` are verified.